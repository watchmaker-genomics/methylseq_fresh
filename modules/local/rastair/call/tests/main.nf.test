
nextflow_process {

    name "Test Process RASTAIR_CALL"
    script "../main.nf"
    process "RASTAIR_CALL"

    tag "rastair"
    tag "rastair/call"

    test("Rastair Call conversion calls") {

        when {
            process {
                """
                input[0] =  [
                    [ id:'test' ], // meta map
				    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.methylated.sorted.bam', checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test' ], // meta map
				    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.methylated.sorted.bam.bai', checkIfExists: true)
                ]
                input[2] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[3] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta.fai', checkIfExists: true)
                ]
                input[4] = [
                    [ id:'test' ], // meta map
                    val(0)
                ]
                input[5] = [
                    [ id:'test' ], // meta map
                    val(0)
                ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                // I need to define the snapshot comparison
                // { assert snapshot(
                //     process.out.txt,
                //     process.out.versions
                //     ).match()
                // }
            )
        }

    }

}
