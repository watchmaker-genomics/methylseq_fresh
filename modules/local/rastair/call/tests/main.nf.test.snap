{
    "Rastair Call conversion calls": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.rastair_call.txt:md5,7508cfdd0045c688f82e4cb4f1d37373"
                ]
            ],
            [
                "versions.yml:md5,f30c6afb2c776f0dd9830769a10b667d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-08-26T13:20:12.548689"
    }
}